src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionRepository.java

```diff
@@ -115,10 +115,13 @@ private JsonObject mapToJson(PatronSessionRecord patronSessionRecord) {
   public CompletableFuture<Result<MultipleRecords<PatronSessionRecord>>> findPatronActionSessions(
     String patronId, PatronActionType actionType, int limit) {
 
-    final Result<CqlQuery> patronIdQuery = exactMatch(PATRON_ID, patronId);
-    final Result<CqlQuery> actionTypeQuery = exactMatch(ACTION_TYPE, actionType.getRepresentation());
+    Result<CqlQuery> cqlQueryResult = exactMatch(PATRON_ID, patronId);
+    if(!PatronActionType.ALL.equals(actionType)) {
```

Comment:
I might be tempted to extract a method for this check. Also the whitespace needs amending.

```suggestion
    if (notAllPatronActionType(actionType)) {
```

Reply:
Done